prusti-launch/src/bin/prusti-rustc.rs

```diff
@@ -42,33 +42,36 @@ fn process(mut args: Vec<String>) -> Result<(), i32> {
     let compiler_bin = prusti_sysroot.join("bin");
     let compiler_lib = prusti_sysroot.join("lib");
 
-    let prusti_home = prusti_driver_path
-        .parent()
-        .expect("Failed to find Prusti's home");
-
     let mut cmd = Command::new(&prusti_driver_path);
 
     add_to_loader_path(vec![compiler_lib, compiler_bin, libjvm_path], &mut cmd);
 
-    prusti_launch::set_environment_settings(&mut cmd, &current_executable_dir, &java_home);
+    prusti_launch::set_environment_settings(&mut cmd, &prusti_home, &java_home);
 
     // Setting RUSTC_WRAPPER causes Cargo to pass 'rustc' as the first argument.
     // We're invoking the compiler programmatically, so we ignore this
-    if !args.is_empty() && Path::new(&args[0]).file_stem() == Some("rustc".as_ref()) {
+    let cargo_invoked = !args.is_empty() && Path::new(&args[0]).file_stem() == Some("rustc".as_ref());
+    if cargo_invoked {
         args.remove(0);
     }
 
-    // filter out the `prusti-contracts` dependency (if present), because we will
```

Comment:
@JonasAlaif In case you get strange compile errors later on, the check for removing a potential prior existing `--extern prusti_contracts={target_dir}/libprusti_contracts.rlib` flag may need to be left intact here. Rustc is a little weird, that when you specify different versions of the same external crate more than once explicitly, then it sometimes selects the wrong version, leading to a compilation error. This happens, because Cargo also adds such a flag when the `prusti-contracts` crate is specified as a dependency already in `Cargo.toml` - and then `prusti-rustc` adds it again. `prusti-tests/tests/cargo_verify/test_no_std/README.md` is also handy for troubleshooting compilation errors, should you encounter any :-)